cupy/cuda/compiler.py

```diff
@@ -145,6 +145,17 @@ def _get_arch():
         return min(arch, _nvrtc_max_compute_capability)
 
 
+def _get_arch_for_options(arch=None, jitify=False):
+    # This is needed to differentiate between
+    # compute or sm depending on the CUDA version
+    # needed to ensure backwards compatibility with nvrtc
+    if arch is None:
+        arch = _get_arch()
```

Comment:
This method does not always return the real device architecture. If we hit `_nvrtc_max_compute_capability` we should use `compute_*` instead.